tests/scale/virt-api.go

```diff
@@ -49,22 +89,38 @@ var _ = Describe("[sig-compute] virt-api scaling", decorators.SigCompute, func()
 		By("Finding out nodes count")
 		nodes, err := virtClient.CoreV1().Nodes().List(context.Background(), v1.ListOptions{})
 		Expect(err).ToNot(HaveOccurred())
-
 		numberOfNodes := len(nodes.Items)
+		expectedResult := calcExpectedReplicas(numberOfNodes)
 
 		Eventually(func() int32 {
-			By("Finding out virt-api replica number")
-			kv := util.GetCurrentKv(virtClient)
-			Expect(kv).ToNot(BeNil())
+			return getApiReplicas(virtClient, expectedResult)
+		}, 1*time.Minute, 5*time.Second).Should(Equal(calcExpectedReplicas(numberOfNodes)), "number of virt API should be as expected")
+	})
 
-			apiDeployment, err := virtClient.AppsV1().Deployments(kv.GetNamespace()).Get(context.Background(), components.VirtAPIName, v1.GetOptions{})
-			Expect(err).ToNot(HaveOccurred())
+	It("[Serial]virt-api replicas should be determined by patch if exist", func() {
+		By("Finding out nodes count")
+		nodes, err := virtClient.CoreV1().Nodes().List(context.Background(), v1.ListOptions{})
+		Expect(err).ToNot(HaveOccurred())
 
-			By("Expecting number of replicas to be as expected")
-			Expect(apiDeployment.Spec.Replicas).ToNot(BeNil())
+		numberOfNodes := len(nodes.Items)
+		expectedResult := calcExpectedReplicas(numberOfNodes)
+		expectedResult += 1
```

Comment:
You can avoid getting the number of nodes in this test, since you are manually patching the spec. IOW, you can use a simple `const` reducing pressure on the api and computation time.

Reply:
Done